From 23f2b45b0d152c115efbcdb72f37c2a75351bbf6 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 10 Jun 2010 13:56:39 +0000
Subject: Show a microphone icon even if the stream has no ID

Just because the stream has no ID doesn't mean that we shouldn't
show the microphone icon.
---
diff --git a/gnome-volume-control/src/gvc-applet.c b/gnome-volume-control/src/gvc-applet.c
index 8de0f43..1b311e4 100644
--- a/gnome-volume-control/src/gvc-applet.c
+++ b/gnome-volume-control/src/gvc-applet.c
@@ -94,8 +94,10 @@ maybe_show_status_icons (GvcApplet *applet)
                         const char *id;
 
                         id = gvc_mixer_stream_get_application_id (s);
-                        if (id == NULL)
-                                continue;
+                        if (id == NULL) {
+                                show = TRUE;
+                                break;
+                        }
 
                         if (!g_str_equal (id, "org.gnome.VolumeControl") &&
                             !g_str_equal (id, "org.PulseAudio.pavucontrol")) {
--
cgit v0.8.3.1
